File: object-oriented-programming/final-game/python/player_class.py
# Isaac Computer Science
# Usage licensed under the Open Government Licence v3.0
# BETA VERSION

from hand_class import Hand

class Player:

    def __init__(self, given_name):
        self.__name = given_name
        self.__score = 0
        self.__hand = Hand()

    def get_name(self):
        return self.__name

    def get_hand(self):
        return self.__hand

    def get_score(self):
        return self.__score

    def set_score(self, new_score):
        self.__score = new_score
    

# This code will run if this file is executed directly
# (i.e. not called by another program)

if __name__ == '__main__':    
    player1 = Player("Leona") # Instantiate an example Player object

File: object-oriented-programming/encapsulation/c-sharp/player_class.cs
/*
Isaac Computer Science
Usage licensed under the Open Government Licence v3.0

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;
using System.IO;

namespace IsaacCodeSamples
{
    class Player
    {

        private string name;
        private int score;

        // Constructor method
        public Player(string givenName) {
            name = givenName;
            score = 0;
        }
        
        public getName() {
            return name;
        }
        
        public getScore() {
            return score;
        }
        
        public setScore(int newScore) {
            score = newScore;
        }
        
    }
    
    
    class Testing
    {
        // The Main method is the entry point for all C# programs
        public static void Main()
        {
            // Instantiate a new player object
            Player player1 = new Player("Leona");
            Console.WriteLine($"The player's name is: {player1.GetName()}");
            Console.WriteLine($"The player's score is: {player1.GetScore()}");
            player1.setScore(20);
            Console.WriteLine($"The player's score is now: {player1.GetScore()}");
        }
    }
    
    
}
